src/README.md

```diff
@@ -0,0 +1,27 @@
+## Welcome to the Babel Codebase
+
+Babel is broken down into three parts:
+
+- Parsing
+- Transformation
+- Generation
+
+**Parsing** is the process of turning a piece of code into an AST (Abstract
+Syntax Tree) which is a tree-like object of nodes that _describes_ the code.
+This makes it easier to transform the code over direct string manpulation.
+
+**Transformation** is the process of taking an AST and manipulating it into
+a new one. For example, Babel might take an ES2015 ArrowFunction and transform
```

Comment:
@benjamingr I recently got the ball rolling on some internals documentation at [`/doc`](https://github.com/babel/babel/tree/master/doc). I think it'd make sense to put a lot of the info there and refer to it from the `README`s and source code comments, but I think the `README`s are a great idea, for now at least.